### PR TITLE
Add `eks.amazonaws.com/role-arn` annotation

### PR DESCRIPTION
**Pull request by DmitryDodzin**
Add `sa.roleArn` value that maps to `eks.amazonaws.com/role-arn` annotation on service account.

relevant for metalbear-co/mirrord#2293.

### CODE REVIEW

**Review by meowjesty — APPROVED** [review]
:+1:

File: mirrord-operator/README.md [hunk 1]
```diff
@@ -20,3 +20,65 @@ If you have a certificate license (usually part of Enterprise offering) you can:
...
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::$account_id:oidc-provider/$oidc_provider"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "$oidc_provider:aud": "sts.amazonaws.com",
+          "$oidc_provider:sub": "system:serviceaccount:$namespace:$service_account"
```
Comment (t4lz) [inline mirrord-operator/README.md:69]:
```suggestion
          "${oidc_provider}:aud": "sts.amazonaws.com",
          "${oidc_provider}:sub": "system:serviceaccount:$namespace:$service_account"
```
For me, the resulting json was messed up without curly brackets here.

File: mirrord-operator/README.md [hunk 2]
```diff
@@ -20,3 +20,65 @@ If you have a certificate license (usually part of Enterprise offering) you can:
...
+        }
+      }
+    }
+  ]
+}
+EOF
+
+aws iam create-role --role-name mirrord-operator-role --assume-role-policy-document file://trust-relationship.json --description "Role for SQS splitting for mirrord-operator"
+
+aws iam attach-role-policy --role-name mirrord-operator-role --policy-arn=arn:aws:iam::$account_id:policy/mirrord-operator-role
```
Comment (t4lz) [inline mirrord-operator/README.md:79]: When I ran this I got
```
An error occurred (NoSuchEntity) when calling the AttachRolePolicy operation: Policy arn:aws:iam::<redacted_by_t4lz>:policy/mirrord-operator-role does not exist or is not attachable.
```

So for my testing I just attached `mirrorrd-operator-role` to the preexisting `AmazonSQSFullAccess` policy (with ARN `arn:aws:iam::aws:policy/AmazonSQSFullAccess`). I don't know if that's a valid solution for our users as well, or if the limitation to a specific account id [in our custom policy](https://github.com/metalbear-co/charts/pull/46/files#diff-8956cd3c524bf05bc926c83fa9d3ae32b07607e7603a9134a57dcb8fb3f83b0eR42) is important.

**Comment by aviramha** [comment]
@Razz4780 @DmitryDodzin @t4lz do we need this?

**Comment by Razz4780** [comment]
> @Razz4780 @DmitryDodzin @t4lz do we need this?

Don't think so, I've seen arn role both in `mirrord operator setup` and in the new chart. @t4lz ? :eyes:

**Comment by DmitryDodzin** [comment]
> @Razz4780 @DmitryDodzin @t4lz do we need this?

Well we can slim up the doc section because it's mostly copy-paste from [aws-docs](https://docs.aws.amazon.com/eks/latest/userguide/associate-service-account-role.html) but the `.Values.sa.roleArn` can be useful in CI setting instead of introducing outside patch to link the arn

**Comment by t4lz** [comment]
I based  on this branch, so these commits are already merged (but I did change the README).
I am confused as to why the diff still shows [stuff](https://github.com/metalbear-co/charts/pull/46/files#diff-4699d97b43c0f321d9dbb2127cfff932a97702572521c3437402068bbb063395R4-R7) that's [already in `main`](https://github.com/metalbear-co/charts/blob/21359e486825c127b577a6fa56cda54e3b5df642/mirrord-operator/templates/service-account.yaml#L4-L7).

**Pull request closed without merging**